IonDotnet/Internals/Text/SystemTextReader.cs

```diff
@@ -453,7 +453,12 @@ public override SymbolToken SymbolValue()
             if (_v.TypeSet.HasFlag(ScalarType.Int) && !_v.TypeSet.HasFlag(ScalarType.String))
             {
                 //lookup symbol string from sid
-                _v.AddString(GetSymbolTable().FindKnownSymbol(_v.IntValue));
+                var text = GetSymbolTable().FindKnownSymbol(_v.IntValue);
+                if (text == null && (_v.IntValue > GetSymbolTable().MaxId + 1 || _v.IntValue < 0))
```

Comment:
Why the +1? The MaxId should be the highest allowable symbol ID; anything above that should be an error.

Reply:
It's because of the [implementation of SymbolToken.cs](https://github.com/amzn/ion-dotnet/blob/master/IonDotnet/SymbolToken.cs#L43).
The comment above it mentions that the actual value is shifted by 1

Reply:
You are right. Fixed and pushed.
Thanks.